src/components/shared/Visualizer.vue

```diff
@@ -151,6 +137,47 @@ export default {
     },
   },
 
+  methods: {
+    highlight(key) {
+      this.highlighted.push(key);
+    },
+    getSVGParams(key, position, keys) {
+      return {
+        width: this.settings.keyCellWidth + (key.toString().length - 1) * 3,
+        height: this.settings.keyCellHeight,
+        x: position * (this.settings.keyCellWidth + (key.toString().length - 1) * 3)
+                - ((this.settings.keyCellWidth + (key.toString().length - 1) * 3) / 2) * (keys.length),
+        y: -this.settings.keyCellHeight / 2,
+        style: this.highlighted.includes(key.toString()) ? this.settings.rectStyles.highlighted : this.settings.rectStyles.plain,
```

Comment:
I think in the leaves, or every key, should have an index ' i ' or ' pos ' that represent the moment it was added. 
So the style should be like this:

`style: this.current === key.pos ? this.settings.rectStyles.highlighted : this.settings.rectStyles.plain`

Reply:
I don't see why we should do this. Care to ellaborate? The highlighted function was a little tool I was planning to use in animations.

Reply:
So there can be more than one leave highlighted by step ? Or the same leave highlighted in more than one step?
If is just one leave by step, this.current is directly related with a single leave so it's logical to make the comparison like I said.